Alignment with upstream

RomanNikitenko opened the pull request:
### What does this PR do?
- Rebase against the upstream
- We still can not align Node.js version with upstream as `20.18.1` version is not available in the images that we use to prepare `che-code` assembly
- Downgrade `vscode/proxy-agent` version to `0.28.0`, see `code/package.json` and `code/remote/package.json`. 
The reason is: `0.29.0` version requires  `20.18.1` version of the Node.js - it is still not available in the `ubi` images
- Downgrade `@playwright/test` - `muscl`-based assembly has a problem with the new version of the `@playwright/test` dep.

### What issues does this PR fix?
<!-- Please include any related issue from the Eclipse Che repository (or from another issue tracker). -->
https://github.com/eclipse-che/che/issues/23310

### How to test this PR?
- PR jobs should be green
- testing of any use cases is useful as we pick up a lot of commits from the upstream

You can use the following links to start a workspace for testing:
- `ubi8-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi8-based
- `ubi8-nodejs-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi8-nodejs-based
- `ubi9-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi9-based
- `ubi9-nodejs-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi9-nodejs-based


### Does this PR contain changes that override default upstream Code-OSS behavior?
- [x] the PR contains changes in the [code](https://github.com/che-incubator/che-code/tree/main/code) folder (you can skip it if your changes are placed in a che extension )
- [x] the corresponding items were added to the [CHANGELOG.md](https://github.com/che-incubator/che-code/blob/main/.rebase/CHANGELOG.md) file
- [ ] rules for automatic `git rebase` were added to the [.rebase](https://github.com/che-incubator/che-code/tree/main/.rebase) folder

vitaliy-guliy commented:
The first test of the terminal highlighted the problem with a huge delay appeared after command execution. Terminal title is blinking from time to time


https://github.com/user-attachments/assets/bf2830a1-3602-4659-a5de-5ab07f761a41

RomanNikitenko commented:
@vitaliy-guliy 
thanks for reporting it!

will try to reproduce the problem...

dependabot[bot] committed:
Bump undici from 7.2.0 to 7.2.3

Bumps [undici](https://github.com/nodejs/undici) from 7.2.0 to 7.2.3.
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v7.2.0...v7.2.3)

---
updated-dependencies:
- dependency-name: undici
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

benibenj committed:
Fix deletion decoration logic in inline diff view (#238377)

fix deletion decoration

jrieken committed:
wip - inline chat via editing session

benibenj committed:
Merge Go To and Accept in gutter menu (#238446)

closes https://github.com/microsoft/vscode-copilot/issues/11923

jrieken committed:
remove `chat.editing.alwaysSaveWithGeneratedChanges` setting and things around it (#238447)

fyi @isidorn

jrieken committed:
Merge branch 'main' into joh/inlineChatEdits

aeschli committed:
Merge pull request #238393 from rcjsuen/icon-extension-point-typo-fix

Fix typo in the help text of the icon extension point

lszomoru committed:
Git - add caching for commit details (#238451)

jrieken committed:
Merge pull request #238422 from tmm1/tmm1/ts-boilerplate-more

build: update to include more tsc boilerplate

lszomoru committed:
Git - enable git blame status bar entry by default (#238454)

lszomoru committed:
StatusBar - keep disposing a status bar entry on the extension host side (#238453)

* StatusBar - keep disposing a status bar entry on the extension host side

* Update src/vs/workbench/api/common/extHostStatusBar.ts

Co-authored-by: Benjamin Pasero <benjamin.pasero@microsoft.com>

* Update src/vs/workbench/api/common/extHostStatusBar.ts

Co-authored-by: Benjamin Pasero <benjamin.pasero@microsoft.com>

* Fix argument names

---------

Co-authored-by: Benjamin Pasero <benjamin.pasero@microsoft.com>

mjbvz committed:
Merge pull request #236640 from mjbvz/passive-penguin

Switch to use nodenext module resolution

mjbvz committed:
Remove unused classification test (#237989)

Remove unused test file

This test has been commented out for 5 years now

meganrogge committed:
show spec info w priority over path (#238408)

sandy081 committed:
make output parsing performant by exposing log entries in the output channel model (#238459)

Tyriar committed:
Upgrade to viewport render strategy for large files

Tyriar committed:
Viewport rendering strategy up to 2000 cols

Instead of dealing with scroll offset for now, just increase column cap significantly

benibenj committed:
Improve border color for inline edits (#238461)

better side by side border color

lszomoru committed:
Git - simplify git blame caching (#238462)

* Git - simplify git blame caching

* 💄 remove code that is not needed

anthonykim1 committed:
Comment out env for bash until fix  (#238463)

comment out env for bash until fix

alexr00 committed:
Tokenize viewport first (#238386)

* Tokenize viewport first

* Actually don't block

bpasero committed:
chat setup - do not bring up confirm dialogs when reloading or closing window (#238468)

jakebailey committed:
Remove paths from tsconfig.base.json

mjbvz committed:
Merge pull request #238475 from jakebailey/remove-tsconfig-paths

Remove paths from tsconfig.base.json

legomushroom committed:
add links provider for prompt files (#238427)

* [prompt links]: add link background decorations

* [prompt links]: cleanup and move files around

* [prompt links]: add II of PromptLinkProvider, ObjectCache and TrckedDisposable utilities

* [prompt links]: add all missing unit tests

* [prompt links]: remove `TextModelDecoratorsProvider`

* [prompt links]: address PR feedback

Tyriar committed:
Move rasterizer ownership up, share between strats

tmm1 committed:
build: switch `build/tsconfig.json` to `module: nodenext` (#238426)

* build/tsconfig: switch to module: nodenext for bun compat

* build: rewrite imports for nodenext compat

* build: re-generate

Tyriar committed:
Rebuild bind buffer when viewport size changes

Tyriar committed:
Tweak initial and increment buffer values

Tyriar committed:
Clean up

Tyriar committed:
Merge branch 'main' into tyriar/227108_viewport_render_strat

legomushroom committed:
[instructions]: remove enable/disable eye button (#238490)

Tyriar committed:
Merge pull request #238487 from microsoft/tyriar/227108_viewport_render_strat

Editor GPU: Add viewport render strategy

bpasero committed:
Copilot setup: support for ghe.com accounts

fix microsoft/vscode-copilot#11890
fix /issues/238013

congyuandong committed:
fix: remove duplicate `!**/*.mk` entry in dependenciesSrc (#236683)

joyceerhl committed:
fix: expand chat/edits empty selection attachments to the whole line (#238499)

bpasero committed:
chat setup - force sign-in when picked (#238502)

lszomoru committed:
Git - handle local tracking branches (#238503)

bpasero committed:
fix: update sign-in error message to reflect correct provider name (#238504)

amunger committed:
fix editable check (#238482)

* fix editable check

* Revert "fix: set _lastFocusedWidget as undefined on widget blur (#234610)"

This reverts commit acc7cf2c73361360267890d99b4818f7cba47131.

connor4312 committed:
chat: fix confirmation cutting off code part (#238509)

Closes https://github.com/microsoft/vscode-copilot/issues/12031

connor4312 committed:
tools: fix edit tool not saving new files before running (#238515)

Fixes https://github.com/microsoft/vscode-copilot/issues/12058

mjbvz committed:
Skip image resize if we don't need to

If the image is under 2048 we can just return the original data

mjbvz committed:
Don't block pastes resolving copy data that we don't care about

mjbvz committed:
Merge pull request #238518 from mjbvz/managing-anglerfish

Skip image resize for small images

mjbvz committed:
Remove logs

meganrogge committed:
fix terminal suggest layout issue (#238466)

fix  #238464

mjbvz committed:
Merge pull request #238521 from mjbvz/remaining-mink

Don't block pastes resolving copy data that we don't care about

legomushroom committed:
integrate instructions into the Edits experience (#238486)

* [edits]: integrate instructions into the Edits experience

* [edits]: update preconditions for the chat execute actions

* [edits]: cleanup

* Update src/vs/workbench/contrib/chat/browser/chatWidget.ts

Co-authored-by: Joyce Er <joyce.er@microsoft.com>

* [edits]: refactor instruction attachments to chat variable conversion logic

* [edits]: chat variable entries for instruction references now different for `child` and `root` references

* [edits]: update variable ID to be unique

---------

Co-authored-by: Joyce Er <joyce.er@microsoft.com>

bpasero committed:
Deep link to Copilot Chat fails if `?referrer` param is added (fix microsoft/vscode-copilot#12053) (#238528)

joaomoreno committed:
fix build (#238530)

* fix build

* fix build

* one more fix

joaomoreno committed:
fix flakiness in extension smoke tests (#238537)

sandy081 committed:
do not make request for extensions with invalid ids (#238535)

alexr00 committed:
Use token store to know what needs to be erefreshed (#238539)

joaomoreno committed:
fix build again (#238545)

bpasero committed:
notifications - adjust positioning (#238546)

When the center opens, it should show notifications exactly where the toast was

sandy081 committed:
Show dialog when installing an extension from a publisher for the first time (#238540)

* #215527 Notify users when installing an extension from a publisher for the first time

* feedback

* polish

* trust publishers post installing

* add verify publisher link

* tweak wording

* tweak wording

* add quotes

* add telelemtry

lszomoru committed:
Git - handle merge conflict refs (#238557)

bpasero committed:
Remove TypeScript ignore comments from upload-sourcemaps files (#238556)

sandy081 committed:
update distro (#238559)

lszomoru committed:
Git - select the correct object for a merge conflict ref (#238560)

bpasero committed:
debt - reduce telemetry use in my components (#238565)

* debt - reduce telemetry use in my components

* debt - reduce telemetry use in my components

sandy081 committed:
use aka link (#238566)

jrieken committed:
inline chat edits

jrieken committed:
undo change

benibenj committed:
Remove NES accepted decoration effect (#238575)

* remove NES accepted decoration effect

* :lipstick:

jrieken committed:
tweaks

dependabot[bot] committed:
Bump undici from 7.2.0 to 7.3.0 in /remote (#238563)

Bumps [undici](https://github.com/nodejs/undici) from 7.2.0 to 7.3.0.
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v7.2.0...v7.3.0)

---
updated-dependencies:
- dependency-name: undici
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sandy081 committed:
support filter by category in all logs (#238582)

meganrogge committed:
add terminal suggest status bar to display actions (#238508)

amunger committed:
set word wrap on correct element (#238493)

* set word wrap on correct element

* update test

jrieken committed:
put toggle icon into navigate group

jrieken committed:
Merge branch 'main' into joh/inlineChatEdits

benibenj committed:
Fix inline completion reuse (#238587)

* #12094

* fix https://github.com/microsoft/vscode-copilot/issues/12094

benibenj committed:
Support showing multi line changes as overlay (#238585)

* multi line replacement

* fix test?

meganrogge committed:
apply font, styling for terminal suggest widget (#238593)

lszomoru committed:
SCM - fix graph hover pills (#238602)

legomushroom committed:
[instructions]: update chat variable IDs creation logic (#238606)

[instructions]: update chat variable IDs creation logic to account for non prompt file references

justschen committed:
add references to `ChatReferenceBinaryData`  (#238516)

* add reference to binary data api

* remove comment

* fix tests

* fix tests try #2

* more testing

* commenting out for testing

* add back history filter

* add under isimage check

* some fixes

* make sure to dispose

Yoyokrazy committed:
Fix URI check for nb inline values (#238613)

fix uri condition

meganrogge committed:
use `terminalShellType` for terminal suggest (#238381)

benibenj committed:
Add wiggle animation on hover for inline edits (#238590)

* wiggle on hover

* :lipstick:

Yoyokrazy committed:
proper cancellation source per nb inline value controller (#238624)

proper cancellation source per inline value controller

rzhao271 committed:
Rename keyMatchSize to keyMatchScore (#238625)

osortega committed:
Persisted storage for replace widget

osortega committed:
Merge pull request #238629 from microsoft/osortega/persisted-storage-replace-widget

Persisted storage for replace widget

HD787 committed:
fixed lazy client

Tyriar committed:
Initialize shellType state after reconnect (#238588)

meganrogge committed:
adjust terminal suggest font, line height when they change (#238611)

sunnylost committed:
fix(settings-editor): ensure the width of the key name does not shrink (#229919)

roblourens committed:
Check api proposal for tool request ID (#238634)

roblourens committed:
Add dropdown for agent mode picker (#238633)

* toggle blue

* Use CheckboxActionViewItem

* Try dropdown

* Polish

* Undo checkbox changes

HD787 committed:
correct conditions

deepak1556 committed:
fix: corrupted vsce dotnet binaries in rpm package (#238636)

jrieken committed:
fix tests

jrieken committed:
Merge pull request #238576 from microsoft/joh/inlineChatEdits

Joh/inlineChatEdits

bpasero committed:
chat setup - cleanup (#238641)

sandy081 committed:
take all profiles for populating trusted publishers (#238638)

alexdima committed:
Fix multiple issues with edits on the last line (#238647)

* Fix multiple issues with edits on the last line

* Fixes for rendering an insertion after the last line

lszomoru committed:
SCM - accessibility help dialog (#238645)

* Initial stub for SCMAccessibilityHelp and SCMAccessibilityHelpContentProvider

* Add content

jrieken committed:
inline chat fixes (#238649)

* workaround weird CSS padding
* fix strings#rcut
* tweak cancel and toggle keybindings

alexdima committed:
Rename the action (#238650)

jrieken committed:
fixes https://github.com/microsoft/vscode-copilot/issues/11860 (#238651)

lszomoru committed:
Accessibility - renamed IAccessibleViewImplentation to IAccessibleViewImplementation (#238652)

benibenj committed:
Add code overlay option for inline suggestions (#238654)

overlay with view zones

alexdima committed:
Increase the weight of Escape (#238655)

alexr00 committed:
Fix bad parsing resuming in tree sitter parser (#238642)

* Fix bad parsing resuming in tree sitter parser

* Don't reset parse when not initial parse

* Don't set tokens when there aren't any

* Remove logging note

benibenj committed:
Choose line replacement or side-by-side view based on editor width (#238659)

* pick linereplcement or side by side depending on editor width

* do not enable overflow

alexr00 committed:
Bumpt distro (#238664)

isidorn committed:
add commithash to telemetry for alignment

isidorn committed:
Merge pull request #238674 from microsoft/isidorn/decent-hyena

add commithash to telemetry for alignment

lszomoru committed:
GitHub - fix event handled (#238676)

benibenj committed:
Fix whitespace insertion rendered as deletion bug (#238677)

whitespace insertion rendered as deletion bug fix

Tyriar committed:
Editor GPU: Pass content width through to scroll bar

Fixes #233790

benibenj committed:
Properly trim prefixes in inline edits (#238672)

proper prefix trim

Tyriar committed:
Merge pull request #238679 from microsoft/tyriar/233790_scroll

Editor GPU: Pass content width through to scroll bar

eleanorjboyd committed:
update distro

benibenj committed:
Add support for multi-line insertions with Ghost Text (#238685)

* multi line insertions with Ghost Text

* :lipstick:

eleanorjboyd committed:
Merge pull request #238686 from microsoft/shivering-crow

 update distro

benibenj committed:
Remove unused color (#238690)

remove dead code

connor4312 committed:
chat: invoke save participants only when accepting changes (#238628)

- Don't trigger save participants in edits from the edit file tool.
  Fixes the bad undo stack we discussed this morning.
- When modifications are accepted, if the file is not dirty, then
  trigger save participants. This lets you still get nice formatting.

karthiknadig committed:
Update distro

roblourens committed:
Rename "Normal" to "Edit" (#238695)

mjbvz committed:
Merge pull request #238630 from HD787/fixLazyClient

enable typescript commands when only config files are open

alexr00 committed:
Fix range end position in piece tree buffer (#238658)

* Fix range end position in piece tree buffer
also fix refresh ranges in token guess

* Simplify first tree sitter tokens

* Compute newRange end for changes in tree sitter

* Fix tests

Tyriar committed:
Sub-pixel offset rendering

Part of #225428

roblourens committed:
Implement past tense invocation message and tooltip for tool calls (#238635)

* Implement past tense invocation message and tooltip for tool calls

* Fix

* Clean up

osortega committed:
Replace history labeling fix

Tyriar committed:
Warm up all sub-pixel offsets

osortega committed:
Merge pull request #238700 from microsoft/osortega/replace-history-labeling-fix

Replace history labeling fix

joyceerhl committed:
fix: remove disposed untitled files from working set (#238704)

bpasero committed:
chat - offer entry to enable more models if Copilot Free user

lramos15 committed:
Add more info to chat install entitlement (#238710)

bpasero committed:
Merge pull request #238711 from microsoft/ben/abstract-leech

chat - offer entry to enable more models if Copilot Free user

connor4312 committed:
tools: allow followup message with implicit cancel on tool confirmation (#238712)

- Make 'requestInProgress' false while waiting for a tool confirmation
- Cancel pending requests when sending a new request via the chat input

connor4312 committed:
chat: disable readonly files for stable for now (#238715)

We're still not confident that this direction is the way to go.

meganrogge committed:
use editor font info for simple completion widget (#238702)

use editor font info

bpasero committed:
splash - include more parts with override support (#238691)

Tyriar committed:
Merge pull request #238699 from microsoft/tyriar/225428_subpixel_offset

Sub-pixel offset rendering

lszomoru committed:
Git - fix autostash to work with staged changes (#238719)

rzhao271 committed:
debt: remove unused filter logging (#238722)

Abrifq committed:
Change `Create New Terminal` to focus accordingly to the terminal location (#237404)

* Change `Create New Terminal` codes to focus accordingly to the profile's location

* Remove showPanel in favor of focusActiveTerminal
in "Create New Terminal (In Active Workspace)"

Co-authored-by: Megan Rogge <merogge@microsoft.com>

---------

Co-authored-by: Megan Rogge <merogge@microsoft.com>

meganrogge committed:
set `top` on resize so widget is positioned correctly (#238723)

fix #235080

meganrogge committed:
Add `~/` completion (#238727)

part of #234352

meganrogge committed:
cleanup suggest settings  (#238709)

sandy081 committed:
fix #238693 (#238726)

rzhao271 committed:
debt: reduce telemetry logging (#238729)

Yoyokrazy committed:
Fix edge cases for nb inline values (#238730)

break, lastCol vs 0, cts token per cell

rzhao271 committed:
feat: comprehensive key match scoring system (#238592)

Also adds a new preview setting that is off by default to prevent more regressions.

rebornix committed:
Avoid fs check on notebook cell uri (#238732)

roblourens committed:
Fix counting chat word loading rate during tool calls (#238731)

The idea was to ignore long pauses, but it wasn't ignoring text edits streaming into the model

rebornix committed:
Prevent file watching for notebook cells (#238733)

No file watching for notebook cell

Yoyokrazy committed:
Adds notebook markup font family support  (#238734)

* edits did this heyo (nb markup font family support)

* setting description includes workbench fallback

TylerLeonhardt committed:
Include an ability to change the default client id (#238736)

Include an ability to change the client id

So our migration is easy to test

legomushroom committed:
unify prompt files configuration setting (#238741)

[prompts]: unify the configuration setting

roblourens committed:
Add toolCalling capability to models (#238737)

* Add toolCalling capability to models
And filter out models that don't support it from the agent

* Fix build

legomushroom committed:
[config]: keep the old config setting name (#238785)

benibenj committed:
Use insert diff background color in side by side view to align with others (#238698)

use same background color in side by side view

bpasero committed:
Allow to determine target URI from untitled editors being saved to disk (fix #238705) (#238790)

bpasero committed:
layout - drop `grid.size` state (#238787)

This key was never updated after the initial startup and thus would always be at the default window size.
With this change, we rely entierly on the computed size of the window using DOM APIs right from the beginning.

bpasero committed:
status - always enable completions entry (#237428) (#238799)

joyceerhl committed:
fix: don't duplicate prompt instructions in chat variables context for edits (#238814)

connor4312 committed:
chat: add a pause button for agent mode (#238735)

* chat: add a pause button for agent mode

There is a new `ChatParticipant.onDidChangePauseState` event that allows
the editor to signal requests should be paused or unpaused.

In agentic mode, a "pause" button is visible in the editor that surfaces
to this event. It's up to the LM to respect it. When paused, actions go
to their "idle" state, so new messages can be written which implicitly
cancel the previous request, following PR earlier today.

![](https://memes.peet.io/img/25-01-b61af958-1390-4143-9e09-cef99b2edb86.png)

This solves Kai's request of wanting to pause and see what the model is
doing before continuing, as well as my desire to be able to interrupt
and hint to the model when it gets off track, although that might
ideally evolve to some other 'hint' vs a cancelled and new request.

PR for copilot coming later this evening.

* allow taking input when paused

joyceerhl committed:
fix: track untitled editor save event in working set (#238816)

bpasero committed:
:up: `@playwright/test@1.50.0` (#238671)

bpasero committed:
chat - make quota indicator top level (#237428) (#238820)

jrieken committed:
edits fixes (#238826)

one inline chat session per file, fix revealing for 1+n requests

bpasero committed:
Clicking on an action in Language Status Indicator causes pins to disappear (fix #238718) (#238788)

benibenj committed:
Update ghost text handling for multi-line rendering (#238827)

dont shift cursor with ghost text multi line rendering

meganrogge committed:
add kb to toggle diff for edits (#238717)

fix #238145

alexr00 committed:
Fix tree sitter freeze and overwriting of scopes (#238829)

Fixes microsoft/vscode-internalbacklog#5307

sandy081 committed:
Fix parsing a log entry with empty lines (#238832)

sandy081 committed:
Introduce Application Machine Scope - fix #235371 (#238825)

* fix #235371

* fix writing application machine setting

* change security.restrictUNCAccess setting also to application machine scope

bpasero committed:
Test failure: workspace.applyEdit drops the TextEdit if there is a RenameFile later #77735 (with opened editor) (fix #238837) (#238838)

benibenj committed:
Update default and public settings for inline suggestions (#238836)

update default and public settings

jrieken committed:
tweak edits progress messages (#238843)

* upper-case edits https://github.com/microsoft/vscode-copilot/issues/11686

* remove window progress when doing edits requests

https://github.com/microsoft/vscode-copilot/issues/11686

aiday-mar committed:
EditContext: Make sure edit context is up to date before using it (#238841)

make sure edit context is up to date before using

bpasero committed:
editors - advise users to check logs for details (fix microsoft/vscode-remote-release#10663) (#238851)

jrieken committed:
fix https://github.com/microsoft/vscode-copilot/issues/11979 (#238853)

Tyriar committed:
Fallback to PriorityTaskQueue on Safari

Fixes #238707

Tyriar committed:
Fix compile issues

bpasero committed:
smoke tests - run with `snapshots: true` for tests that install extensions (#238855)

Tyriar committed:
Merge pull request #238858 from microsoft/tyriar/238707

Fallback to PriorityTaskQueue on Safari

legomushroom committed:
rename `settledAll` methods to `allSettled` for consistency (#238862)

[prompts]: rename `settledAll` methods to `allSettled` for consistency

oxcened committed:
fix: check whether lastFocusedList is valid when assigned (#238765)

benibenj committed:
Enhance NES ghost text with highlighting and bubble (#238863)

* Use ghost text with highlighting and bubble around it

* :lipstick:

sandy081 committed:
Check for deps or packs in trusting publisher dialog (#238867)

sandy081 committed:
wording tweak (#238876)

rzhao271 committed:
feat: expose tags in configuration schema (#238738)

Fixes: https://github.com/microsoft/vscode/issues/231917

meganrogge committed:
fix case when font > 16 and line height <= 1 for terminal suggest (#238865)

amunger committed:
disable executing REPL history with enter (#238877)

alexr00 committed:
Fix incorrect and too small tree sitter changed ranges (#238870)

* Fix tree sitter edit apply bug

* Only use named nodes for tree sitter changes

* Fix test

connor4312 committed:
lists: fix bugs around user selection in lists (#238879)

- Only start scroll animation on a non-empty selection
- Stop scroll animation as soon as the mouse is released

aeschli committed:
apply in editor: show target in hover (#238881)

connor4312 committed:
testing: allow expanding test output diff when inline history is visible (#238883)

Fixes #237850

meganrogge committed:
rm webpack config from terminal-suggest (#238874)

rzhao271 committed:
fix: unclear setting indicator label content

Fixes #235139

rzhao271 committed:
Clarify setting types (#238885)

mjbvz committed:
Merge pull request #238889 from microsoft/rzhao271/indicator-labels

fix: unclear setting indicator label content

meganrogge committed:
tweak condition to terminal suggest hiding bug (#238894)

tweak condition to fix bug

aeschli committed:
[json] update service (#238898)

meganrogge committed:
Set `isFile/isDirectory` for completion items (#238900)

Set isFile/isDirectory for completion items

Co-authored-by: Daniel Imms <2193314+Tyriar@users.noreply.github.com>

connor4312 committed:
eng: rev TPN for 1.97 (#238902)

benibenj committed:
Improve UI rendering and accessibility features (#238903)

* small UI rendering improvements

* :lipstick:

roblourens committed:
Clear edit session when toggling agent mode (#238904)

* Clear edit session when toggling agent mode
And add agent-specific welcome message

* Message

rzhao271 committed:
Fix #197755 (#238905)

rzhao271 committed:
fix: Settings editor left pane shows up too often (#238906)

Fixes #196468

roblourens committed:
Add setting: chat.agent.maxRequests (#238909)

TylerLeonhardt committed:
Don't overlap window controls (#238910)

Fixes https://github.com/microsoft/vscode/issues/237626

roblourens committed:
Ignore file edit limit when in agent mode (#238914)

legomushroom committed:
trigger chat input re-layout when number of instruction  attachments changes (#238915)

[prompts]: trigger chat input re-layout when number of instruction attachments changes

legomushroom committed:
fixes the error/warning CSS classname of instruction attachment (#238917)

[prompts]: fixes the error/warning CSS classname of instruction attachment

RomanNikitenko committed:
chore: Downgrade vscode/proxy-agent dep as 20.18.1 Node.js is not available

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

meganrogge committed:
Revert "rm webpack config from terminal-suggest" (#238981)

Revert "rm webpack config from terminal-suggest (#238874)"

This reverts commit b54c554bb2ea20e567c6347097f8e9fdbac166ca.

atreids committed:
chore: fix typo in VSIX progress notification (#238845)

alexr00 committed:
Fix some places tree sitter throws and can cause a freeze (#238987)

isidorn committed:
add aiplatform module to look for

isidorn committed:
Merge pull request #238993 from microsoft/isidorn/boring-galliform

add aiplatform module to look for

meganrogge committed:
Make terminal suggest provider enabled by default (#238869)

fix  #238081

legomushroom committed:
remove the `.prompt.md` file extension in the attached instruction label (#239010)

[instructions]: remove the `.prompt.md` file extension in the attached instruction label

osortega committed:
Fix for no results found message on ai results

legomushroom committed:
rename `instructions` to `prompt` in the UI (#239013)

[instructions]: rename `instructions` to `prompt` in the UI

osortega committed:
Merge pull request #239023 from microsoft/osortega/ai-results-message-fix

Fix for no results found message on ai results

connor4312 committed:
debug: bump js-debug to 1.97 (#239034)

legomushroom committed:
[prompts]: fix typos (#239014)

mjbvz committed:
Explain how multiple providers work for paste and drop

Fixes #238978

legomushroom committed:
updates the prompt label in the attachments list (#239045)

[prompts]: change `Prompt` to `Prompt..` in the attachments list

legomushroom committed:
add docs link to empty prompt selection dialog (#239050)

* [prompts]: add docs link to the empty prompts selection dialog

* [prompt]: fix a typo in localization string ID

* [prompts]: refactor to reduce number of function arguments

* Update src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts

Co-authored-by: Harald Kirschner <hkirschner@microsoft.com>

---------

Co-authored-by: Harald Kirschner <hkirschner@microsoft.com>

legomushroom committed:
publish configuration setting for prompt files (#239053)

* [prompts]: make the configuration setting public(experimental)

* [prompts]: update config setting name

* [prompts]: cleanup

* [prompts]: fix localization compilation error

roblourens committed:
Change "Explore New Features" to "Show Release Notes" (#239058)

roblourens committed:
Don't toggle agent mode when clicking the selected option in the dropdown (#239059)

mjbvz committed:
Merge pull request #239042 from mjbvz/generous-armadillo

Explain how multiple providers work for paste and drop

benibenj committed:
Fix line replacement view width computation (#239064)

line replace view width computation fix

benibenj committed:
Fix insertion view height and position calculations (#239065)

insertion view fixes

jrieken committed:
fix https://github.com/microsoft/vscode-copilot/issues/12393 (#239066)

jrieken committed:
fixes https://github.com/microsoft/vscode/issues/238926 (#239067)

justschen committed:
filter out images from history (#239069)

jrieken committed:
fix https://github.com/microsoft/vscode/issues/238925 (#239070)

jrieken committed:
workaround for https://github.com/microsoft/vscode-copilot/issues/11811 (#239072)

benibenj committed:
Fix NES freezes window issue (#239074)

fix NES freezes window

alexr00 committed:
Improve description for comments.thread.confirmOnCollapse (#239075)

Fixes microsoft/vscode-pull-request-github#6618

benibenj committed:
Fix gutter icon positioning over minimap (#239076)

fix https://github.com/microsoft/vscode-copilot/issues/12501

jrieken committed:
avoid cyclic dependency between `ChatEditorOverlayController` and `ChatEditorController` (#239077)

fixes https://github.com/microsoft/vscode-copilot/issues/12252

sandy081 committed:
fix #238952 (#239079)

RomanNikitenko committed:
Rebase against the upstream ea254757de5d467b73578f9dfb2f5633a6e548da
vscode-upstream-sha1: ea254757de5d467b73578f9dfb2f5633a6e548da

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
chore: Update lock file after resolving conflicts

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Downgrade @playwright/test

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

RomanNikitenko committed:
chore: Add info related to rebasing against upstream

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

RomanNikitenko committed:
chore: Merge main to resolve conflicts

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

RomanNikitenko committed:
Merge branch 'main' into alignment-with-upstream
